server/manager/src/main/java/org/apache/accumulo/master/Master.java

```diff
@@ -1178,6 +1178,9 @@ boolean canSuspendTablets() {
       log.info("All metrics modules registered");
     }
 
+    // checking stored user hashes if any of them uses an outdated algorithm
+    security.validateStoredUserCreditentials();
```

Comment:
I am wondering if this is the best place to validate all user creds.  On one hand, it may be better to do this for some situations every time Master starts up but on the other, it may be sufficient and more efficient to only check once on upgrade.

Reply:
I was wondering that, too, but I kind of want users to be annoyed by it more than once, so that they actually log users in to upgrade the hashes. I was also thinking maybe the other route, and ~~nagging~~ *logging* once every few days or once a week on a running system. Once at master startup is probably a good compromise for now, though. We can get more aggressive in future releases, before eventually forcing it at upgrade.

Reply:
Good point.  And we can easily move the check somewhere else in later versions.